src/main/java/com/mongodb/jdbc/MySQLColumnInfo.java

```diff
@@ -0,0 +1,182 @@
+package com.mongodb.jdbc;
+
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.HashSet;
+import java.util.Set;
+import org.bson.BsonType;
+
+public class MySQLColumnInfo implements MongoColumnInfo {
+    public String database;
+    public String table;
+    public String tableAlias;
+    public String column;
+    public String columnAlias;
+    public String bsonType;
+    private static Set<String> nullTypes;
+    private BsonType bsonTypeEnum;
+    private int jdbcType;
+
+    static {
+        nullTypes = new HashSet<String>();
+        nullTypes.add("dbPointer");
+        nullTypes.add("javascript");
+        nullTypes.add("javascriptWithScope");
+        nullTypes.add("maxKey");
+        nullTypes.add("minKey");
+        nullTypes.add("object");
+        nullTypes.add("regex");
+        nullTypes.add("symbol");
+        nullTypes.add("timestamp");
+        nullTypes.add("undefined");
+    }
+
+    public MySQLColumnInfo(
+            String database,
+            String table,
+            String tableAlias,
+            String column,
+            String columnAlias,
+            String bsonType)
+            throws SQLException {
+        this.database = database;
+        this.table = table;
+        this.tableAlias = tableAlias;
+        this.column = column;
+        this.columnAlias = columnAlias;
+        this.bsonType = bsonType;
+    }
+
+    public void init() throws SQLException {
```

Comment:
The POJO codec is not going to call the constructor, so we use this init method to set data that isn't specified by the bson doc.